pkg/accounting/accounting.go

```diff
@@ -216,6 +261,44 @@ func (a *Accounting) getPeerBalance(peer swarm.Address) (*PeerBalance, error) {
 	return peerBalance, nil
 }
 
-func (pb *PeerBalance) freeBalance() int64 {
+// expectedBalance returns the balance we expect to have with a peer if all reserved funds are actually credited
+func (pb *PeerBalance) expectedBalance() int64 {
 	return pb.balance - int64(pb.reserved)
 }
+
+// expectedDebt returns the debt we expect to have with a peer if all reserved funds are actually credited
+func (pb *PeerBalance) expectedDebt() uint64 {
+	expectedBalance := pb.expectedBalance()
+	if expectedBalance >= 0 {
+		return 0
+	}
+	return uint64(-expectedBalance)
+}
+
+// NotifyPayment is called by Settlement when we received payment
+func (a *Accounting) NotifyPayment(peer swarm.Address, amount uint64) error {
+	balance, err := a.getPeerBalance(peer)
+	if err != nil {
+		return err
+	}
+
+	balance.lock.Lock()
+	defer balance.lock.Unlock()
+
+	nextBalance := balance.balance - int64(amount)
+
+	if nextBalance < 0 {
+		return fmt.Errorf("refusing to accept payment which would put us in debt, new balance would have been %d", nextBalance)
```

Comment:
I think this check is too harsh. 
First concern: 
- A node might actually send a higher amount that what we think he should send us, due to differences in balances
Second concern:
- Why shouldn't I just accept a higher payment and go on?

My suggestion would be to: always accept a payment, where we do proper accounting if the payment received makes the balance above zero (but doesn't trigger a payment from our side) and where we don't do "proper" accounting when the payment would trigger us to send a cheque back immediately.

Reply:
It now allows putting us into debt up to the payment tolerance

Reply:
Cool, thanks! This works if the payment tolerance is strictly less than they payment threshold. Do you think there is a need for a hard check that this is the case when we start up the node? Or perhaps only documenting it is enough?
Don't forget to update the PR description to include this too!

Reply:
Don't forget this comment @ralph-pichler